FreeRTOS/Demo/RISC-V-ISP/Makefile

```diff
@@ -0,0 +1,259 @@
+#
+# File: Makefile
+# Copyright (c) 2019, Dornerworks Ltd.
+#
+
+PROJ_NAME ?= simple
+ARCH ?= rv32
+ARCH_XLEN = $(subst rv,,$(ARCH))
+
+BUILD_DIR = ./build/$(ARCH)
+OUT_LIB = $(BUILD_DIR)/libfreertos-hifive$(ARCH_XLEN).a
+
+SDK_DIR = ./freedom-e-sdk
+PLAT_DIR = $(SDK_DIR)/env/freedom-e300-hifive1/
+PLAT_LINK_SCRIPT = flash.lds
+
+LINKER_SCRIPT ?= $(PLAT_DIR)/$(PLAT_LINK_SCRIPT)
+
+include BuildEnvironment.mk
+include FreeRTOS.mk
+-include $(wildcard $(OUTDIR)/*.d)
+# Platform spcific settings for this application _____________________________________________
+
+CFLAGS := $(ISP_CFLAGS)
+CFLAGS += $(CONFIG_CFLAGS)
+ASMFLAGS := $(ISP_ASMFLAGS)
+ASMFLAGS += $(CONFIG_ASMFLAGS)
+LDFLAGS := $(ISP_LDFLAGS)
+
+ifneq ($(filter rv64 rv32,$(ARCH)),"")
+   BASE_ARCH = RISC-V
+endif
+ARCH_PORTABLE_INC = $(FREERTOS_SOURCE_DIR)/portable/GCC/$(BASE_ARCH)/
+ARCH_PORTABLE_SRC = $(FREERTOS_SOURCE_DIR)/portable/GCC/$(BASE_ARCH)/port.c
+ARCH_PORTABLE_ASM = $(FREERTOS_SOURCE_DIR)/portable/GCC/$(BASE_ARCH)/portASM.S
+
+LDFLAGS += -L$(BUILD_DIR)
+LDFLAGS += -Wl,-Map,"$(BUILD_DIR)/$(PROJ_NAME).map"
+
+SDK_SRC := \
+	$(SDK_DIR)/libwrap/sys/_exit.c \
+	$(SDK_DIR)/libwrap/sys/close.c \
+	$(SDK_DIR)/libwrap/sys/execve.c \
+	$(SDK_DIR)/libwrap/sys/fork.c \
+	$(SDK_DIR)/libwrap/sys/fstat.c \
+	$(SDK_DIR)/libwrap/sys/getpid.c \
+	$(SDK_DIR)/libwrap/sys/isatty.c \
+	$(SDK_DIR)/libwrap/sys/kill.c \
+	$(SDK_DIR)/libwrap/sys/link.c \
+	$(SDK_DIR)/libwrap/sys/lseek.c \
+	$(SDK_DIR)/libwrap/sys/openat.c \
+	$(SDK_DIR)/libwrap/sys/puts.c \
+	$(SDK_DIR)/libwrap/sys/printf.c \
+	$(SDK_DIR)/libwrap/sys/read.c \
+	$(SDK_DIR)/libwrap/sys/sbrk.c \
+	$(SDK_DIR)/libwrap/sys/stat.c \
+	$(SDK_DIR)/libwrap/sys/times.c \
+	$(SDK_DIR)/libwrap/sys/unlink.c \
+	$(SDK_DIR)/libwrap/sys/wait.c \
+	$(SDK_DIR)/libwrap/sys/write.c \
+	$(SDK_DIR)/libwrap/misc/write_hex.c \
+	$(SDK_DIR)/env/freedom-e300-hifive1/init.c \
+	$(SDK_DIR)/drivers/plic/plic_driver.c \
+	$(SDK_DIR)/drivers/fe300prci/fe300prci_driver.c
+
+SDK_ASM := \
+	$(SDK_DIR)/env/entry.S \
+	$(SDK_DIR)/env/start.S
+SDK_OBJS := $(patsubst %.c,$(FREERTOS_BUILD_DIR)/%.o,$(notdir $(SDK_SRC)))
+SDK_OBJS += $(patsubst %.S,$(FREERTOS_BUILD_DIR)/%.o,$(notdir $(SDK_ASM)))
+VPATH += \
+	$(SDK_DIR)/env \
+	$(SDK_DIR)/env/freedom-e300-hifive1 \
+	$(SDK_DIR)/libwrap/sys \
+	$(SDK_DIR)/libwrap/misc \
+	$(SDK_DIR)/libwrap/stdlib \
+	$(SDK_DIR)/drivers/plic \
+	$(SDK_DIR)/drivers/fe300prci
+
+PORT_OBJS := $(patsubst %.c,$(FREERTOS_BUILD_DIR)/%.o,$(notdir $(ARCH_PORTABLE_SRC)))
+PORT_OBJS += $(patsubst %.S,$(FREERTOS_BUILD_DIR)/%.o,$(notdir $(ARCH_PORTABLE_ASM)))
+FREERTOS_OBJS += $(PORT_OBJS) $(SDK_OBJS)
+VPATH += $(FREERTOS_SOURCE_DIR)/portable/GCC/$(BASE_ARCH)
+
+
+CFLAGS += -DportasmHANDLE_INTERRUPT=$(INTERRUPT_HANDLER)
+
+# Set up application source, include, and object files for compilation: ______________________
+APP_SRC_DIR = .
+APP_SRC = \
+	$(APP_SRC_DIR)/main.c \

```

Comment:
We might as well get rid of the application source if we're making our own demo, since we don't use these